testruby.c

```diff
@@ -1,24 +1,51 @@
 #include <stdio.h>
 #include <stdlib.h>
-int i, M, N, x, y, j, z, var, red, st1, st2, st, red1, red2, l, d;
-char c = 'A', b, pamti, boja;
-char **tablica;
-void regija(int x, int y, char boja, char pamti)
+#include <stdbool.h>
+int M, N;
+char **matrix;
+bool isPixelValid(int x, int y)
 {
-    if (tablica[x][y] == boja)
+    if (x < 0 || x > N - 1 || y < 0 || y > M - 1)
+        return false;
+    else
+        return true;
+}
+char getPixelColor(int x, int y)
+{
+    if (isPixelValid(x, y) == true)
+        return matrix[x][y];
+        else return '\0';

```

Comment:
you do not need an explicit else, if if condition is true function will exit hence to 'return' statement, if the conditiona inside if is false it will proceed to the line 17 which just can be 'return '\0' 

this is called a 'guard clause' https://medium.com/@scadge/if-statements-design-guard-clauses-might-be-all-you-need-67219a1a981a